### PR TITLE
Add failing of edits with edit notes

### PR DESCRIPTION
**Pull request by ghost**
* Refactors the performer edit handling to move logic out of the query builder, similar to scene handling.
* Add `StashBot` user which posts edit notes for failed edits.
* Fails edit when an error is encountered during application.
* Changes system user management to check for root/stashbot on boot every time.
* Adds tooltip for edit status with explanations for failing statuses.
* Removes fail condition when trying to remove a join that no longer exists, or add one that is already there. They are instead ignored.
* Adds `BOT` user role. Not used for anything right now, but might be useful in the future.

### CODE REVIEW

File: pkg/models/translate.go [hunk 1]
```diff
@@ -8,6 +8,16 @@ import (
...
 )
 
+type ErrEditPrerequisiteFailed struct {
+	field    string
+	expected interface{}
+	actual   interface{}
+}
+
+func (e *ErrEditPrerequisiteFailed) Error() string {
+	return fmt.Sprintf("Expected %s to be “**%v**”, but was “**%v**”.", e.field, e.expected, e.actual)
```
Comment (peolic) [inline pkg/models/translate.go]: > Edit application failed:
>
> Prerequisite failed: Expected Details to be “text text text”, but was “****”.

Empty values cause the 'bold' markdown to not render.

Comment (ghost) [reply pkg/models/translate.go]: Thanks, fixed!

File: pkg/user/user.go [hunk 2]
```diff
@@ -502,3 +536,16 @@ func PromoteUserVoteRights(fac models.Repo, userID uuid.UUID, threshold int) err
 
 	return nil
 }
+
+func GetModUser(fac models.Repo) *models.User {
+	if modUser == nil {
+		user, err := fac.User().FindByName("StashBot")
```
Comment (peolic) [inline pkg/user/user.go]:
```suggestion
		user, err := fac.User().FindByName(modUserName)
```

**Review by peolic — APPROVED** [review]

**Comment by peolic** [comment]
Also run UI generate. `RoleEnum` on `globalTypes.ts` is missing the new `BOT` role.

**Pull request merged**